Source/MQTTnet/Internal/BlockingQueue.cs

```diff
@@ -8,7 +8,7 @@ public class BlockingQueue<TItem>
     {
         private readonly object _syncRoot = new object();
         private readonly LinkedList<TItem> _items = new LinkedList<TItem>();
-        private readonly ManualResetEvent _gate = new ManualResetEvent(false);
+        private readonly ManualResetEventSlim _gate = new ManualResetEventSlim(false);
```

Comment:
ManualResetEventSlim is Disposable. So please implement the Dispose interface. Also parent classes might need to call Dispose then as well.

Reply:
Ok, i'll do it later

Reply:
I have some time at the moment. I will do. Hope you did not start already 🙈